packages/clerk-js/src/ui/signUp/SignUpForm.tsx

```diff
@@ -85,14 +87,14 @@ export function SignUpForm({
           </Control>
         )}
 
-        {fields.emailAddress && (
+        {fields.emailAddress && (toggleEmailPhone ? activeCommIdentifierType === 'emailAddress' : true) && (
```

Comment:
Not a fan of this approach. Should we introduce a new show Field and use that instead?

Reply:
The logic has been moved to the place we calculate the active fields to be rendered